lib/application.js

```diff
@@ -592,7 +593,14 @@ app.render = function(name, options, fn){
 
 app.listen = function(){
   var server = http.createServer(this);
-  return server.listen.apply(server, arguments);
+  var args = Array.prototype.slice.call(arguments);
```

Comment:
Neat, I found an even simpler/more robust solution to this: call `.listen` and just see if there is now a `listening` event listener attached, and use that :)